File: atg1.java
import java.util.*;
public class Main {
    public static List<Integer> sortDescending(List<Integer> array) {
    Collections.sort(array, Collections.reverseOrder());
    return array;
}

public static boolean isGoodGraph(List<Integer> array) {
    if (array.get(0) != 1) {
        return false;
    }
    for (int i = 1; i < array.size(); i++) {
        if (array.get(i) != 0) {
            return false;
        }
    }
    return true;
}

public static boolean hasNegativeElements(List<Integer> array) {
    for (int x : array) {
        if (x < 0) {
            return true;
        }
    }
    return false;
}

public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    System.out.print("Zadaj pocet vrcholov: ");
    int numVertices = input.nextInt();
    List<Integer> degrees = new ArrayList<>();
    for (int i = 0; i < numVertices; i++) {
        int degreeNum = i + 1;
        System.out.printf("Zadaj %d. stupen vrchola: ", degreeNum);
        degrees.add(input.nextInt());
    }
    List<Integer> sortedDegrees = sortDescending(degrees);
    System.out.println(sortedDegrees);
    for (int j = 0; j < sortedDegrees.size(); j++) {
        int temp = sortedDegrees.get(j);
        for (int k = 1; k <= temp; k++) {
            sortedDegrees.set(j, 0);
            sortedDegrees.set(j+k, sortedDegrees.get(j+k) - 1);
        }
        sortedDegrees = sortDescending(sortedDegrees);
        if (isGoodGraph(sortedDegrees)) {
            System.out.println("Uspesny graf");
            break;
        }
        if (hasNegativeElements(sortedDegrees)) {
            System.out.println("Neuspesny graf");
            break;
        }
    }
}

File: atg1.py
def zoradPole(pole):
    for i in range(0, len(pole)):    
        for j in range(i+1, len(pole)):    
            if(pole[i] < pole[j]):    
                temp = pole[i]   
                pole[i] = pole[j] 
                pole[j] = temp

def jeDobre(pole):
    jeDobre = False
    for i in range(len(pole)):
        if i == 0:
            if pole[i] == 1:
                for j in range(1, len(pole)):
                    jeDobre = True
                    if pole[j] != 0:
                        jeDobre = False
    return jeDobre

def jeZle(pole):
    jeZle = False
    for i in range(len(pole)):
        if pole[i] < 0:
            jeZle = True
    return jeZle

def main():
    pocetVrcholov = int(input("Zadaj pocet vrcholov: "))
    stupneVrcholov = []
    for i in range(pocetVrcholov):
        pocet = i + 1
        stupneVrcholov.append(int(input(f"Zadaj {pocet}. stupen  vrchola: ")))
    zoradPole(stupneVrcholov)
    print(stupneVrcholov)
    for j in range(len(stupneVrcholov)):
       
        temp = stupneVrcholov[j]
        for k in range(1,temp+1):
            stupneVrcholov[j] = 0
            stupneVrcholov[j+k] -= 1
        zoradPole(stupneVrcholov)
        if jeDobre(stupneVrcholov):
            print("Uspesny graf")
            break
        if  jeZle(stupneVrcholov):
            print("Neuspesny Graf")
            break
    
main()